hoomd/mpcd/ParticleData.cc

```diff
@@ -742,6 +749,23 @@ void mpcd::ParticleData::setMass(Scalar mass)
     #endif // ENABLE_MPI
     }
 
+/*!
+ * \param N number of particles on that rank
+ * \return N_global_new (sum of all the N on every rank)
+ */
+unsigned int mpcd::ParticleData::calculateN_global(unsigned int m_N)
+    {
+    unsigned int N_global_new = m_N;
+    #ifdef ENABLE_MPI
+    if (m_exec_conf->getNRanks() > 1)
+        {
+        MPI_Allreduce(&m_N, &N_global_new, 1, MPI_UNSIGNED, MPI_SUM, m_exec_conf->getMPICommunicator());
```

Comment:
This is an in-place reduction
```suggestion
        MPI_Allreduce(MPI_IN_PLACE, &N_global, 1, MPI_UNSIGNED, MPI_SUM, m_exec_conf->getMPICommunicator());
```